extensions/import/src/wizard/pages/prosePreviewPage.ts

```diff
@@ -107,22 +106,36 @@ export class ProsePreviewPage extends ImportPage {
 	}
 
 	async onPageEnter(): Promise<boolean> {
-		this.loading.loading = true;
 		let proseResult: boolean;
 		let error: string;
-		try {
-			proseResult = await this.handleProse();
-		} catch (ex) {
-			error = ex.toString();
+		const enablePreviewFeatures = vscode.workspace.getConfiguration('workbench').get('enablePreviewFeatures');
+		if (this.model.newFileSelected) {
+			this.loading.loading = true;
+			try {
+				proseResult = await this.handleProse();
+			} catch (ex) {
+				error = ex.toString();
+				this.instance.wizard.message = {
+					level: azdata.window.MessageLevel.Error,
+					text: error
+				};
+			}
+			this.model.newFileSelected = false;
+			this.loading.loading = false;
 		}
-
-		this.loading.loading = false;
-		if (proseResult) {
-			await this.populateTable(this.model.proseDataPreview, this.model.proseColumns.map(c => c.columnName));
+		if (!this.model.newFileSelected || proseResult) {
+			const tempTable = JSON.parse(JSON.stringify(this.model.proseDataPreview));
```

Comment:
Parsing a stringified object should just return the original object, is there a reason you're doing this here?